### ISSUE TITLE
feature request: using provided `metric_function` as the objective for HO 

### ISSUE THREAD

**Issue by krzyzinskim** [ModelOriented/forester#69]
The `metric_function` can't be used as a loss function in hyperparameter optimization - it would be useful to be able to use this function in this process as well.
